paasta_tools/cli/cmds/validate.py

```diff
@@ -507,16 +507,24 @@ def _get_comments_for_key(data: CommentedMap, key: Any) -> Optional[str]:
     # this is a little weird, but ruamel is returning a list that looks like:
     # [None, None, CommentToken(...), None] for some reason instead of just a
     # single string
-    raw_comments = [
-        comment.value for comment in data.ca.items.get(key, []) if comment is not None
-    ]
+    # Sometimes ruamel returns a recursive list of CommentTokens as well that looks like
+    # [None, None, [CommentToken(...),CommentToken(...),None], CommentToken(...), None]
+    def _flatten_comments(comments):
+        for comment in comments:
+            if comment is None:
+                continue
+            if isinstance(comment, list):
+                yield from _flatten_comments(comment)
+            else:
+                yield comment.value
+
+    raw_comments = [*_flatten_comments(data.ca.items.get(key, []))]
```

Comment:
While doing further testing I've found ruamel to be very inconsistent with its return behaviour. For example when using the a base template such as below will return both the comments in line 3/4
```yaml
_base_template: &base_template
  registrations: ["experiment_allocation.main"]
  # this comment will be returned by raumel
  cpus: 1  # this comment will ALSO be returned by raumel

main:
  <<: *base_template
  deploy_group: prod.non_canary
...
```

however when not using a base template such as it will NOT return the comment in line 2 but will only return comment in line 3.
```yaml
main:
  # this comment will be ignored
  cpus: 1 #only this comment wil be returned by raumel
```

However in both examples any comments written below the cpus line will be returned.